collect_app/src/main/java/org/odk/collect/android/widgets/SelectMultiMinimalWidget.java

```diff
@@ -0,0 +1,83 @@
+package org.odk.collect.android.widgets;
+
+import android.content.Context;
+
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.core.model.data.SelectMultiData;
+import org.javarosa.core.model.data.helper.Selection;
+import org.odk.collect.android.R;
+import org.odk.collect.android.activities.FormEntryActivity;
+import org.odk.collect.android.formentry.questions.QuestionDetails;
+import org.odk.collect.android.fragments.dialogs.SelectMinimalDialog;
+import org.odk.collect.android.fragments.dialogs.SelectMultiMinimalDialog;
+import org.odk.collect.android.utilities.StringUtils;
+import org.odk.collect.android.utilities.WidgetAppearanceUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SelectMultiMinimalWidget extends SelectMinimalWidget {
+    private List<Selection> selectedItems;
+
+    public SelectMultiMinimalWidget(Context context, QuestionDetails prompt) {
+        super(context, prompt);
+        selectedItems = getFormEntryPrompt().getAnswerValue() == null
+                ? new ArrayList<>() :
+                (List<Selection>) getFormEntryPrompt().getAnswerValue().getValue();
+        updateAnswerLabel();
+    }
+
+    @Override
+    protected void showDialog() {
+        SelectMultiMinimalDialog dialog = new SelectMultiMinimalDialog(getSavedSelectedItems(), WidgetAppearanceUtils.isFlexAppearance(getFormEntryPrompt()), WidgetAppearanceUtils.isAutocomplete(getFormEntryPrompt()), getChoicesRecyclerViewProps());
```

Comment:
How did you decide what goes in the props record vs. not? Why not include flex and autocomplete appearances?

Reply:
Because flex and autocomplete appearances are used in `SelectMinimalDialog` but props are passed to an adapter
https://github.com/getodk/collect/pull/3964/files#diff-c84dd1657c1e1200edf5f3ca7b8f062aR23 that was my reasoning.

Reply:
I see, that makes sense. You know, the more I think about this the more it feels like a pattern we don't want to introduce. It's not very Java/Android idiomatic and I don't think there's much advantage to it. It's ok to have long parameter lists! You could also consider something like a `SelectQuestionPrompt` object that wraps the form entry prompt and booleans for the different appearances. When that object is created with a `FormEntryPrompt`, the appearance is read and sanitized once and all the booleans are set. That also has the advantage of moving away from the static `WidgetAppearanceUtils`.

Reply:
I agree it's not good I didn't like the long list of parameters and wanted to improve it somehow but ended up with another approach that is not perfect. I'll take another look maybe we can improve it or just pass all parameters directly as you said it's not that bad.